### PR TITLE
Clarify where DNS resolution works.

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

File: cluster/addons/dns/README.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ crashes or scheduling changes).  This maps well to DNS, which has a long
 history of clients that, on purpose or on accident, do not respect DNS TTLs
 (see previous remark about Pod IPs changing).
 
+## Where does resolution work?
+Service DNS names can be resolved within Pods but not outside the cluster,
+or on a node outside any container (e.g. ssh to node).
```
Comment (thockin) [inline cluster/addons/dns/README.md]: you can do DNS from nodes but you need to specify the DNS server explicitly.

Comment (thockin) [inline cluster/addons/dns/README.md]: e.g. dig @10.0.0.10 mysvc.ns.svc.cluster.local

**Comment by thockin** [comment]
1 not then LGTM

**Comment by ArtfulCoder** [comment]
It might help to clarify that dig @10.0.0.10 will not work from the master.

File: cluster/addons/dns/README.md (hunk 1 above)
Comment (erictune) [inline cluster/addons/dns/README.md]: fixed.

**Comment by erictune** [comment]
Actually, I don't see a succint way to explain all the ways in which Tim's suggestion might or might not work.  Leaving it out.

**Comment by erictune** [comment]
@ArtfulCoder PTAL

**Comment by erictune** [comment]
Per offline convo, I will update to state that when a pod has hostNet, we don't automatically add search and nameserver entries to /etc/resolv.conf.

File: cluster/addons/dns/README.md [hunk 2]
```diff
@@ -13,6 +13,13 @@ crashes or scheduling changes).  This maps well to DNS, which has a long
 history of clients that, on purpose or on accident, do not respect DNS TTLs
 (see previous remark about Pod IPs changing).
 
+## Where does resolution work?
+
+Cluster DNS resolution standard methods (e.g. [`gethostbyname
+(3)`](http://linux.die.net/man/3/gethostbyname), `dig`) should always work
+within a Pod.  Outside of a Pod, there is no such guarantee: environments will
```
Comment (ArtfulCoder) [inline cluster/addons/dns/README.md]: I would leave the "Outside of a Pod.." statement out completely, or replace it with "outside of a Pod, the FQDN needs to be used and the cluster nameserver needs to be explicitly specified during DNS lookups"

**Pull request merged**